### ISSUE TITLE
Historical lookup of numbers

### ISSUE THREAD

**Issue by pbelmans** [stacks/stacks-website#105]
Even though one should only use tags and not the actual numbering to refer to lemmas etc. the use of numbers is unfortunately widespread. To prevent references from becoming meaningless after a while, we could implement a system in which it is possible to look up the tag for a certain number at a certain date. People like to refer to "lemma 38.4.1" more than they like to refer to "tag 0SP4", and I don't know what publishers think with regards to citing the Stacks project in publications. But judging on the use links to the Stacks project on MathOverflow, Math.StackExchange, and arXiv preprints, it might not be a bad idea to implement such a thing.

So whether it is desirable I don't know yet, but it is certainly possible.

**Comment by aisejohan** [comment]
The problem with this idea is that not all versions were actually a version on the web. Ignoring that it could be done, but it would mean a lot of computation, as I do not see how to figure out what the numbers are, except by compiling the stacks project.... Moreover, not every version compiles...

Interesting idea though.

**Comment by pbelmans** [comment]
Yes, it's not perfect, but it would come with a disclaimer that it's the web version. Otherwise one has to compile every single commit of the Stacks project and store its numbers :). Or maybe we could use some cloud computing to compile 4608 versions of the Stacks project (whenever they compile of course).

More seriously: whenever the update script is run, it could store the then used numbers together with a date, making a historical lookup possible.

It's technically possible. I'm not sure I want to implement it soon.
